src/Api.php

```diff
@@ -234,74 +323,85 @@ public function delete($pattern, $callable = null)
+        $f = function () use ($model) {
             $args = func_get_args();
 
             if (is_callable($model)) {
-                $model = call_user_func_array($model, $args);
+                $model = $this->call($model, $args);
             }
 
             return $model;
-        });
+        };
+        $this->get($pattern, $f);
 
         // GET :id - one record
-        $this->get($pattern.'/:id', function () use ($model) {
+        $f = function () use ($model) {
             $args = func_get_args();
             $id = array_pop($args); // pop last element of args array, it's :id
 
             if (is_callable($model)) {
-                $model = call_user_func_array($model, $args);
+                $model = $this->call($model, $args);
             }
 
+            // limit fields
+            $model->onlyFields($this->getAllowedFields($model, 'read'));
+
             return $model->load($id)->get();
-        });
+        };
+        $this->get($pattern.'/:id', $f);
 
+        // POST :id - update one record
         // PATCH :id - update one record (same as POST :id)
-        $this->patch($pattern.'/:id', function () use ($model) {
+        $f = function () use ($model) {
             $args = func_get_args();
             $id = array_pop($args); // pop last element of args array, it's :id
 
             if (is_callable($model)) {
-                $model = call_user_func_array($model, $args);
+                $model = $this->call($model, $args);
             }
 
-            return $model->load($id)->save($this->requestData)->get();
-        });
+            // limit fields
+            $model->onlyFields($this->getAllowedFields($model, 'modify'));
+            $model->load($id)->save($this->requestData);
+            $model->onlyFields($this->getAllowedFields($model, 'read'));
 
-        // POST :id - update one record
-        $this->post($pattern.'/:id', function () use ($model) {
+            return $model->get();
+        };
+        $this->patch($pattern.'/:id', $f);
+        $this->post($pattern.'/:id', $f);
+
+        // POST - insert new record
+        $f = function () use ($model) {
             $args = func_get_args();
-            $id = array_pop($args); // pop last element of args array, it's :id
 
             if (is_callable($model)) {
-                $model = call_user_func_array($model, $args);
+                $model = $this->call($model, $args);
             }
 
-            return $model->load($id)->save($this->requestData)->get();
-        });
+            // limit fields
+            $model->onlyFields($this->getAllowedFields($model, 'modify'));
+            $model->unload()->save($this->requestData);
+            $model->onlyFields($this->getAllowedFields($model, 'read'));
+
+            return $model->get();
+        };
+        $this->post($pattern, $f);
 
         // DELETE :id - delete one record
-        $this->delete($pattern.'/:id', function () use ($model) {
+        $f = function () use ($model) {
```

Comment:
unused $f ?

Reply:
oh its used below, n/p.

Reply:
yeah it's to avoid code duplication for post/patch methods and for consistency i set $f for other methods too.